pkg/applicationserver/distribution/subscription_set.go

```diff
@@ -97,7 +98,7 @@ func (s *subscriptionSet) Subscribe(ctx context.Context, protocol string, ids *t
 // Publish publishes the upstream traffic to the subscribers.
 func (s *subscriptionSet) Publish(ctx context.Context, up *ttnpb.ApplicationUp) error {
 	ctxUp := &io.ContextualApplicationUp{
-		Context:       ctx,
+		Context:       s.FromRequestContext(ctx),
```

Comment:
Is this the reason to pass the `Component` all the way here?

I think we get too high coupling of `Component`; why not declare a receiving interface?

Reply:
I couldn't think of a good name for this particular decoupling - `RequestDecoupler` sounds a bit off. Thoughts ?

Reply:
Yeah why not?

Reply:
Done :+1: